server/internal/core/application/covenantless.go

```diff
@@ -948,89 +958,125 @@ func (s *covenantlessService) listenToScannerNotifications() {
 
 	mutx := &sync.Mutex{}
 	for vtxoKeys := range chVtxos {
-		go func(vtxoKeys map[string]ports.VtxoWithValue) {
+		go func(vtxoKeys map[string][]ports.VtxoWithValue) {
 			vtxosRepo := s.repoManager.Vtxos()
 			roundRepo := s.repoManager.Rounds()
 
-			for _, v := range vtxoKeys {
-				// redeem
-				vtxos, err := vtxosRepo.GetVtxos(ctx, []domain.VtxoKey{v.VtxoKey})
-				if err != nil {
-					log.WithError(err).Warn("failed to retrieve vtxos, skipping...")
-					continue
-				}
+			for _, keys := range vtxoKeys {
+				for _, v := range keys {
+					// redeem
+					vtxos, err := vtxosRepo.GetVtxos(ctx, []domain.VtxoKey{v.VtxoKey})
+					if err != nil {
+						log.WithError(err).Warn("failed to retrieve vtxos, skipping...")
+						continue
+					}
 
-				vtxo := vtxos[0]
+					vtxo := vtxos[0]
 
-				if vtxo.Redeemed {
-					continue
-				}
+					if vtxo.Redeemed {
+						continue
+					}
 
-				if err := s.repoManager.Vtxos().RedeemVtxos(
-					ctx, []domain.VtxoKey{vtxo.VtxoKey},
-				); err != nil {
-					log.WithError(err).Warn("failed to redeem vtxos, retrying...")
-					continue
-				}
-				log.Debugf("vtxo %s redeemed", vtxo.Txid)
+					if err := s.repoManager.Vtxos().RedeemVtxos(
+						ctx, []domain.VtxoKey{vtxo.VtxoKey},
+					); err != nil {
+						log.WithError(err).Warn("failed to redeem vtxos, retrying...")
+						continue
+					}
+					log.Debugf("vtxo %s:%d redeemed", vtxo.Txid, vtxo.VOut)
 
-				if !vtxo.Spent {
-					continue
-				}
+					if !vtxo.Spent {
+						continue
+					}
 
-				log.Debugf("fraud detected on vtxo %s", vtxo.Txid)
+					log.Debugf("fraud detected on vtxo %s:%d", vtxo.Txid, vtxo.VOut)
```

Comment:
let's move all the following code to a `reactToFraud` function to properly use `defer mutx.Unlock()` instead of having it repeated in several lines.